docs/get_started/centos_setup.md

```diff
@@ -0,0 +1,164 @@
+
+Note: If you are not familiar with CentOS or building progress, we encourage you to [install or deploy MXNet environment on Ubuntu](http://mxnet.io/get_started/ubuntu_setup.html). 
+
+## Minimum Requirements
+Make sure you have the root permission, and `yum` is properly installed.
+
+To install MXNet on CentOS, you must have the following:
+
+1. gcc, g++ (4.8 or later)
+2. python2, python-numpy, python-pip, clang
+3. graphviz, jupyter (pip or yum install)
+4. OpenBLAS
+5. CUDA for GPU
+6. opencv (do not use yum to install opencv, some shared libs may not be installed) and cmake
+
+### Basic packages
+Make sure your machine is connected to Internet. A few installations need to download (`git clone` or `wget`) some packages from Internet.
+
+```bash
+	# Install gcc-4.8/make and other development tools
+	sudo yum install -y gcc
+	sudo yum install -y gcc-c++
+	sudo yum install -y clang
+
+	# Install Python, Numpy, pip and set up tools.
+	sudo yum groupinstall -y "Development Tools"
+	sudo yum install -y python27 python27-setuptools python27-tools python-pip
+	sudo yum install -y python27-numpy 
+
+	# install graphviz, jupyter
+	sudo pip install graphviz
+	sudo pip install jupyter
+```
+### Install OpenBLAS
+Note that OpenBLAS can be replaced by other BLAS libs, e.g, Intel MKL.
+
+```bash
+	# Install OpenBLAS at /usr/local/openblas
+	git clone https://github.com/xianyi/OpenBLAS
+	cd OpenBLAS
+	make -j $(($(nproc) + 1))
+	sudo make PREFIX=/usr/local install
+	cd ..
+```
+### Install CUDA for GPU
+If you plan to build with GPU, you need to set up the environment for CUDA and CUDNN.
+
+First, download and install [CUDA 8 toolkit](https://developer.nvidia.com/cuda-toolkit).
+
+Then download [cudnn 5](https://developer.nvidia.com/cudnn).
+
+Unzip the file and change to the cudnn root directory. Move the header and libraries to your local CUDA Toolkit folder:
+
+```bash
+    tar xvzf cudnn-8.0-linux-x64-v5.1-ga.tgz
+    sudo cp -P cuda/include/cudnn.h /usr/local/cuda/include
+    sudo cp -P cuda/lib64/libcudnn* /usr/local/cuda/lib64
+    sudo chmod a+r /usr/local/cuda/include/cudnn.h /usr/local/cuda/lib64/libcudnn*
+    sudo ldconfig
+```
+### Install opencv for Computer Vision and Image Augmentation
+The Open Source Computer Vision (OpenCV) library contains programming functions for computer vision and image augmentation. For more information, see [OpenCV](https://en.wikipedia.org/wiki/OpenCV).
+
+```bash
+	sudo yum install -y cmake
+	# Install OpenCV at /usr/local/opencv
+	git clone https://github.com/opencv/opencv
+	cd opencv
+	mkdir -p build
+	cd build
+	cmake -D BUILD_opencv_gpu=OFF -D WITH_EIGEN=ON -D WITH_TBB=ON -D WITH_CUDA=OFF -D WITH_1394=OFF -D CMAKE_BUILD_TYPE=RELEASE -D CMAKE_INSTALL_PREFIX=/usr/local ..
+	sudo make PREFIX=/usr/local install
+```
+
+## Install MXNet
+
+### Generate libmxnet.so
+After installing the dependencies, use the following command to pull the MXNet source code from GitHub.
+
+```bash
+    # Get MXNet source code
```

Comment:
Comment can be - Download MXNet source code to ~/mxnet directory
to help reader know where it will get downloaded.